src/api/core/warren/experience_index/models.py

```diff
@@ -0,0 +1,150 @@
+"""Experience Index SQL and Pydantic Models."""
+from datetime import datetime, timezone
+from enum import CONTINUOUS, IntEnum, StrEnum, auto, unique, verify
+from typing import List, Optional, Union
+
+from pydantic import Json
+from pydantic.main import BaseModel
+from sqlalchemy import Column, Enum
+from sqlalchemy.types import JSON, DateTime
+from sqlmodel import Field, Relationship, SQLModel
+
+from warren.fields import IRI
+
+
+@unique
+class Kind(StrEnum):
+    """wip."""
+
+    ISPARTOF = auto()
+    HASPART = auto()
+    ISVERSIONOF = auto()
+
+
+class Relation(SQLModel, table=True):
+    """wip."""
+
+    source_id: IRI = Field(
+        foreign_key="experience.id", primary_key=True, description="wip"
+    )
+    target_id: IRI = Field(
+        foreign_key="experience.id", primary_key=True, description="wip"
+    )
+    kind: Kind = Field(sa_column=Column(Enum(Kind)), description="wip")
+    description: Json = Field(sa_column=Column(JSON), description="wip")
+
+
+@unique
+class Structure(StrEnum):
+    """Enumeration of Experience Structures."""
+
+    ATOMIC = auto()
+    COLLECTION = auto()
+    NETWORKED = auto()
+    HIERARCHICAL = auto()
+    LINEAR = auto()
+
+
+@unique
+@verify(CONTINUOUS)
+class AggregationLevel(IntEnum):
+    """Enumeration of Experience Aggregation Levels."""
+
+    ONE = auto()
+    TWO = auto()
+    THREE = auto()
+    FOUR = auto()
```

Comment:
Maybe there is a simpler way :sweat_smile: 

```suggestion
class AggregationLevel(IntEnum):
    """Enumeration of Experience Aggregation Levels."""

    ONE = 1
    TWO = 2
    THREE = 3
    FOUR = 4
```

Reply:
The decorators add extra validation to the Enum for prevention against potential regressions. I prefer keeping at least the @unique decorator. It doesn't hurt readability and makes the code more reliable.

Reply:
`auto()` + `IntEnum` generate integers for each value, you find this approach less readable?

Reply:
I find it too convoluted, less explicit, too machine-trustworthy. But you can keep it that way, it's not a bid deal. :wink:

Reply:
Alright, I have aligned the enum declaration with the `Structure` declaration.